From 597f299a33672e83d41c51051d6ae33a8595a3f8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 2 May 2013 22:26:21 +0100
Subject: [PATCH] MobileVLCKit: link against freetype

---
 MobileVLCKit/MobileVLCKit.xcconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/MobileVLCKit/MobileVLCKit.xcconfig b/MobileVLCKit/MobileVLCKit.xcconfig
index a5ebc87..c31e2e9 100644
--- a/MobileVLCKit/MobileVLCKit.xcconfig
+++ b/MobileVLCKit/MobileVLCKit.xcconfig
@@ -19,6 +19,7 @@ VLC_CONTRIB_DIR=$VLC_INSTALL_DIR/contribs
 AVCODEC=$(VLC_CONTRIB_DIR)/lib/libavutil.a ${VLC_CONTRIB_DIR}/lib/libavcodec.a $(VLC_CONTRIB_DIR)/lib/libavformat.a $(VLC_CONTRIB_DIR)/lib/libavresample.a $(VLC_CONTRIB_DIR)/lib/libswscale.a $(VLC_CONTRIB_DIR)/lib/libgsm.a
 DVBPSI=$(VLC_CONTRIB_DIR)/lib/libdvbpsi.a
 FLAC=$(VLC_CONTRIB_DIR)/lib/libFLAC.a
+FREETYPE=$(VLC_CONTRIB_DIR)/lib/libfreetype.a
 LIVE555=$(VLC_CONTRIB_DIR)/lib/libUsageEnvironment.a $(VLC_CONTRIB_DIR)/lib/libBasicUsageEnvironment.a $(VLC_CONTRIB_DIR)/lib/libgroupsock.a $(VLC_CONTRIB_DIR)/lib/libz.a $(VLC_CONTRIB_DIR)/lib/libliveMedia.a
 LUA=$(VLC_CONTRIB_DIR)/lib/liblua.a
 MKV=$(VLC_CONTRIB_DIR)/lib/libebml.a $(VLC_CONTRIB_DIR)/lib/libmatroska.a
@@ -29,4 +30,4 @@ PICS=$(VLC_CONTRIB_DIR)/lib/libpng.a $(VLC_CONTRIB_DIR)/lib/libopenjpeg.a
 SPEEX=$(VLC_CONTRIB_DIR)/lib/libspeex.a $(VLC_CONTRIB_DIR)/lib/libspeexdsp.a
 XML=$(VLC_CONTRIB_DIR)/lib/libxml2.a
 
-OTHER_LIBTOOLFLAGS=$(AVCODEC) $(DVBPSI) $(FLAC) $(LIVE555) $(LUA) $(MKV) $(MODPLUG) $(OGG) $(OPUS) $(PICS) $(SPEEX) $(XML) $(VLC_INSTALL_DIR)/lib/vlc/libcompat.a  $(VLC_INSTALL_DIR)/lib/libvlc.a $(VLC_INSTALL_DIR)/lib/libvlccore.a $(VLC_PLUGINS_LDFLAGS) $(OTHER_CFLAGS)
+OTHER_LIBTOOLFLAGS=$(AVCODEC) $(DVBPSI) $(FLAC) $(FREETYPE) $(LIVE555) $(LUA) $(MKV) $(MODPLUG) $(OGG) $(OPUS) $(PICS) $(SPEEX) $(XML) $(VLC_INSTALL_DIR)/lib/vlc/libcompat.a  $(VLC_INSTALL_DIR)/lib/libvlc.a $(VLC_INSTALL_DIR)/lib/libvlccore.a $(VLC_PLUGINS_LDFLAGS) $(OTHER_CFLAGS)
-- 
1.7.12.4 (Apple Git-37)

