From 3c224ced6b5de1c3ddc93545dfd0b61182ef6520 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 10 Aug 2012 16:02:33 +0200
Subject: [PATCH 4/8] deinterlace: added a hack to work-around a libtool bug

---
 modules/video_filter/Modules.am |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/video_filter/Modules.am b/modules/video_filter/Modules.am
index 183009a..e5facc1 100644
--- a/modules/video_filter/Modules.am
+++ b/modules/video_filter/Modules.am
@@ -1,3 +1,7 @@
+# When using a specific AS tool instead of gcc libtool goes mad
+# without this. (This is the case with iOS).
+LIBTOOL=@LIBTOOL@ --tag=CC
+
 SOURCES_mosaic = mosaic.c mosaic.h
 SOURCES_transform = transform.c
 SOURCES_invert = invert.c
-- 
1.7.10.2 (Apple Git-33)

