From 0b437a1b4aec64429b9f8fc759e1834110ba2a36 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 8 Aug 2013 22:12:57 +0200
Subject: [PATCH 04/17] extras/ios: enable zvbi compilation

---
 extras/package/ios/build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index 39d90bd..8155062 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -190,7 +190,7 @@ fi
     --disable-sdl \
     --disable-SDL_image \
     --disable-iconv \
-    --disable-zvbi \
+    --enable-zvbi \
     --disable-kate \
     --disable-caca \
     --disable-gettext \
-- 
1.7.12.4 (Apple Git-37)

