From 4dc4c72304446e736a0e209f2fcc180331bb724a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 13 Sep 2012 01:36:39 +0200
Subject: [PATCH 06/10] contrib/ebml: fix CXXFLAG visibility hack

---
 contrib/src/ebml/rules.mak | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/contrib/src/ebml/rules.mak b/contrib/src/ebml/rules.mak
index f4f01c7..862553c 100644
--- a/contrib/src/ebml/rules.mak
+++ b/contrib/src/ebml/rules.mak
@@ -16,9 +16,10 @@ libebml: libebml-$(EBML_VERSION).tar.bz2 .sum-ebml
 	$(MOVE)
 
 # libebml requires exceptions
-EBML_EXTRA_FLAGS = CXXFLAGS="${CXXFLAGS} -fexceptions" CPPFLAGS=""
 ifdef HAVE_IOS
-EBML_EXTRA_FLAGS += " -fvisibility=hidden"
+EBML_EXTRA_FLAGS = CXXFLAGS="${CXXFLAGS} -fexceptions -fvisibility=hidden" CPPFLAGS=""
+else
+EBML_EXTRA_FLAGS = CXXFLAGS="${CXXFLAGS} -fexceptions" CPPFLAGS=""
 endif
 
 .ebml: libebml
-- 
1.7.12.4 (Apple Git-37)

