From 5840a570de7bb46601d9ec1b43d01a9d4875fa59 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Sun, 21 Jul 2013 18:18:33 +0200
Subject: [PATCH 08/15] freetype: fix corner-case crash

---
 modules/text_renderer/freetype.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/text_renderer/freetype.c b/modules/text_renderer/freetype.c
index 9a00739..c7f3c1f 100644
--- a/modules/text_renderer/freetype.c
+++ b/modules/text_renderer/freetype.c
@@ -825,6 +825,8 @@ static char *iOSFake_Select( filter_t *p_filter, bool b_monospace, bool b_bold,
                                       CFSTR("OpenSans-Regular.ttf") : CFSTR("SourceCodePro-Regular.ttf"),
                                       NULL,
                                       NULL);
+    if (!fileURL)
+        return NULL;
     CFStringRef urlString = CFURLCopyFileSystemPath(fileURL, kCFURLPOSIXPathStyle);
     CFRelease(fileURL);
 
-- 
1.8.3.4 (Apple Git-47)

