From 32011da1639d30338878d02ecbea9bf67bfeb1ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 2 May 2013 20:30:35 +0100
Subject: [PATCH 10/10] ios: enable freetype

---
 extras/package/ios/build.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index 6d1b9c3..fa813c5 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -193,7 +193,6 @@ fi
     --disable-SDL_image \
     --disable-fontconfig \
     --disable-ass \
-    --disable-freetype2 \
     --disable-iconv \
     --disable-fribidi \
     --disable-zvbi \
@@ -215,6 +214,7 @@ fi
     --disable-libmpeg2 \
     --disable-chromaprint \
     --disable-mad \
+    --enable-freetype2 \
     --disable-taglib > ${out}
 
 echo "EXTRA_CFLAGS += ${EXTRA_CFLAGS}" >> config.mak
@@ -317,7 +317,7 @@ ${VLCROOT}/configure \
     --enable-theora \
     --enable-flac \
     --disable-screen \
-    --disable-freetype \
+    --enable-freetype \
     --disable-taglib \
     --disable-mmx \
     --disable-mad > ${out} # MMX and SSE support requires llvm which is broken on Simulator
-- 
1.7.12.4 (Apple Git-37)

