From 2e19907ceb0dd6cb5ab86bb755fefe352689963c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20Ku=CC=88hne?= <fkuehne@videolan.org>
Date: Sun, 28 Jul 2013 19:24:30 +0200
Subject: [PATCH 11/17] extras/package: compile quartztext module for iOS

---
 extras/package/ios/build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index fa99c2d..0181585 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -268,7 +268,7 @@ ${VLCROOT}/configure \
     --enable-ios-vout \
     --enable-ios-vout2 \
     --disable-shared \
-    --disable-macosx-quartztext \
+    --enable-macosx-quartztext \
     --enable-avcodec \
     --enable-mkv \
     --enable-opus \
-- 
1.7.12.4 (Apple Git-37)

