From 87a7999f1fa4583627721751433d85b045fd539a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 27 Feb 2013 21:17:03 +0100
Subject: [PATCH 11/11] vout_ios2: compilation fix

---
 modules/video_output/ios2.m | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/modules/video_output/ios2.m b/modules/video_output/ios2.m
index 9238055..a18f8f2 100644
--- a/modules/video_output/ios2.m
+++ b/modules/video_output/ios2.m
@@ -309,9 +309,8 @@ static int Control(vout_display_t *vd, int query, va_list ap)
 
 static void PictureDisplay(vout_display_t *vd, picture_t *pic, subpicture_t *subpicture)
 {
-    sys->has_first_frame = true;
-
     vout_display_sys_t *sys = vd->sys;
+    sys->has_first_frame = true;
     vout_display_opengl_Display(sys->vgl, &vd->source);
 
     picture_Release(pic);
-- 
1.7.12.4 (Apple Git-37)

