From 316586ace910955aaa800b7883cc941f9d3a9cd9 Mon Sep 17 00:00:00 2001
From: Gleb Pinigin <gpinigin@gmail.com>
Date: Fri, 19 Jul 2013 20:52:51 +0700
Subject: [PATCH] ios: disable VNCServer contrib

---
 extras/package/ios/build.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index c9fea60..bf065b0 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -203,6 +203,7 @@ fi
     --disable-samplerate \
     --disable-goom \
     --disable-gcrypt \
+    --disable-vncserver \
     --disable-gnutls \
     --disable-orc \
     --disable-schroedinger \
-- 
1.7.12.4 (Apple Git-37)

