From 74e06034414965dc7fda4afe9423f8d15ee8ee49 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Thu, 3 Oct 2013 18:28:26 +0200
Subject: [PATCH 13/15] extras: deploy arch option for iOS

---
 extras/package/ios/build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index 25dbf17..92efdb1 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -183,7 +183,7 @@ else
   export ASCPP="xcrun as"
 fi
 
-../bootstrap --host=${TARGET} --build="i686-apple-darwin10" --prefix=${VLCROOT}/contrib/${TARGET}-${ARCH} --disable-gpl \
+../bootstrap --host=${TARGET} --build="i686-apple-darwin10" --prefix=${VLCROOT}/contrib/${TARGET}-${ARCH} --arch=${ARCH} --disable-gpl \
     --disable-disc --disable-sout \
     --disable-sdl \
     --disable-SDL_image \
-- 
1.8.3.4 (Apple Git-47)

