From a6ef28d8028280a4e5a53ca2a47bad045d17b18c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Sun, 21 Jul 2013 18:18:33 +0200
Subject: [PATCH 14/14] freetype: fix corner-case crash

---
 modules/text_renderer/freetype.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/text_renderer/freetype.c b/modules/text_renderer/freetype.c
index e1dfea2..098c67b 100644
--- a/modules/text_renderer/freetype.c
+++ b/modules/text_renderer/freetype.c
@@ -825,6 +825,8 @@ static char *iOSFake_Select( filter_t *p_filter, bool b_monospace, bool b_bold,
                                       CFSTR("OpenSans-Regular.ttf") : CFSTR("SourceCodePro-Regular.ttf"),
                                       NULL,
                                       NULL);
+    if (!fileURL)
+        return NULL;
     CFStringRef urlString = CFURLCopyFileSystemPath(fileURL, kCFURLPOSIXPathStyle);
     CFRelease(fileURL);
 
-- 
1.8.3.1 (Apple Git-46)

