From 782eccd0d1ae98cc9455d44b2aeb20c3b99879c4 Mon Sep 17 00:00:00 2001
From: Gleb Pinigin <gpinigin@gmail.com>
Date: Fri, 17 May 2013 23:13:06 +0700
Subject: [PATCH 15/15] Fix potential crash when sending notification to
 deallocated object

---
 modules/video_output/ios2.m | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/video_output/ios2.m b/modules/video_output/ios2.m
index 0774550..9ecc58c 100644
--- a/modules/video_output/ios2.m
+++ b/modules/video_output/ios2.m
@@ -398,6 +398,7 @@ static void OpenglESSwap(vlc_gl_t *gl)
 
 - (void)dealloc
 {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
     [_eaglContext release];
     [super dealloc];
 }
-- 
1.7.12.4 (Apple Git-37)

