From 9ac6ada49d2f38ae1dde6fcee552a46e252058c5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 30 Oct 2013 20:03:42 +0100
Subject: [PATCH 1/2] contrib/gcrypt: fix compilation for iOS

---
 contrib/src/gcrypt/rules.mak | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/contrib/src/gcrypt/rules.mak b/contrib/src/gcrypt/rules.mak
index 31b930e..7d2688e 100644
--- a/contrib/src/gcrypt/rules.mak
+++ b/contrib/src/gcrypt/rules.mak
@@ -37,6 +37,10 @@ endif
 
 .gcrypt: libgcrypt
 	#$(RECONF)
-	cd $< && $(HOSTVARS) ./configure $(HOSTCONF) $(GCRYPT_CONF)
+ifdef HAVE_IOS
+	cd $< && $(HOSTVARS) CFLAGS="$(CFLAGS) -fheinous-gnu-extensions" ./configure $(HOSTCONF) $(GCRYPT_CONF)
+else
+	cd $< && $(HOSTVARS) CFLAGS="$(CFLAGS) -fheinous-gnu-extensions" ./configure $(HOSTCONF) $(GCRYPT_CONF)
+endif
 	cd $< && $(MAKE) install
 	touch $@
-- 
1.8.3.4 (Apple Git-47)

