From d5a4a989c90fe715589af16d3ef6d15123467114 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Wed, 30 Oct 2013 20:04:22 +0100
Subject: [PATCH 2/2] extras/ios: enable gcrypt as we need it for HLS playback

---
 extras/package/ios/build.sh | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/extras/package/ios/build.sh b/extras/package/ios/build.sh
index 131a025..ff8b9a9 100755
--- a/extras/package/ios/build.sh
+++ b/extras/package/ios/build.sh
@@ -198,7 +198,6 @@ fi
     --disable-sidplay2 \
     --disable-samplerate \
     --disable-goom \
-    --disable-gcrypt \
     --disable-vncserver \
     --disable-gnutls \
     --disable-orc \
@@ -212,7 +211,6 @@ fi
     --enable-ass \
     --disable-fontconfig \
     --disable-gpg-error \
-    --disable-gcrypt \
     --disable-lua \
     --disable-taglib > ${out}
 
@@ -275,7 +273,6 @@ ${VLCROOT}/configure \
     --enable-fribidi \
     --disable-macosx-audio \
     --disable-qt --disable-skins2 \
-    --disable-libgcrypt \
     --disable-vcd \
     --disable-vlc \
     --disable-vlm \
-- 
1.8.3.4 (Apple Git-47)

