From 9e1f00db0c5db45a5c1047445955db97bfde95db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Sun, 15 Dec 2013 17:27:51 +0100
Subject: [PATCH 3/3] Update BWCrashReportTextFormatter for new PLCrashReporter

---
 client/iOS/BWCrashReportTextFormatter.m | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/client/iOS/BWCrashReportTextFormatter.m b/client/iOS/BWCrashReportTextFormatter.m
index 79e4569..ceb725b 100644
--- a/client/iOS/BWCrashReportTextFormatter.m
+++ b/client/iOS/BWCrashReportTextFormatter.m
@@ -148,7 +148,6 @@ static NSInteger binaryImageSort(id binary1, id binary2, void *context) {
             switch (report.systemInfo.architecture) {
                 case PLCrashReportArchitectureARMv6:
                 case PLCrashReportArchitectureARMv7:
-                case PLCrashReportArchitectureARMv7s:
                     codeType = @"ARM";
                     lp64 = false;
                     break;
@@ -173,17 +172,11 @@ static NSInteger binaryImageSort(id binary1, id binary2, void *context) {
     }
 
     {
-        NSString *reportGUID = @"[TODO]";
-        if ([report respondsToSelector:@selector(reportInfo)]) {
-            if (report.hasReportInfo && report.reportInfo.reportGUID != nil)
-                reportGUID = report.reportInfo.reportGUID;
-        }
-      
+
         NSString *hardwareModel = @"???";
         if (report.hasMachineInfo && report.machineInfo.modelName != nil)
             hardwareModel = report.machineInfo.modelName;
 
-        [text appendFormat: @"Incident Identifier: %@\n", reportGUID];
         [text appendFormat: @"CrashReporter Key:   [TODO]\n"];
         [text appendFormat: @"Hardware Model:      %@\n", hardwareModel];
     }
-- 
1.8.3.4 (Apple Git-47)

