Index: GTMOAuth2Authentication.h
===================================================================
--- GTMOAuth2Authentication.h	(revision 131)
+++ GTMOAuth2Authentication.h	(working copy)
@@ -23,7 +23,7 @@
 #if (!TARGET_OS_IPHONE && defined(MAC_OS_X_VERSION_10_11) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11) \
   || (TARGET_OS_IPHONE && defined(__IPHONE_9_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
   #ifndef GTM_USE_SESSION_FETCHER
-    #define GTM_USE_SESSION_FETCHER 1
+    #define GTM_USE_SESSION_FETCHER 0
   #endif
 #endif
 
