Index: GTLService.h
===================================================================
--- GTLService.h	(revision 457)
+++ GTLService.h	(working copy)
@@ -30,13 +30,6 @@
 // GTMSessionFetcher rather than the older GTMHTTPFetcher.  The session
 // fetcher requires iOS 7/OS X 10.9 and supports out-of-process uploads.
 
-#if (!TARGET_OS_IPHONE && defined(MAC_OS_X_VERSION_10_11) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11) \
-  || (TARGET_OS_IPHONE && defined(__IPHONE_9_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
-  #ifndef GTM_USE_SESSION_FETCHER
-    #define GTM_USE_SESSION_FETCHER 1
-  #endif
-#endif
-
 #if !defined(GTL_USE_SESSION_FETCHER) && defined(GTM_USE_SESSION_FETCHER)
   #define GTL_USE_SESSION_FETCHER GTM_USE_SESSION_FETCHER
 #endif  // GTL_USE_SESSION_FETCHER
