diff -ru upnpx-1.2.4/src/upnp/UPnPManager.m upnpx/src/upnp/UPnPManager.m
--- upnpx-1.2.4/src/upnp/UPnPManager.m	2012-03-07 20:53:57.000000000 +0100
+++ upnpx/src/upnp/UPnPManager.m	2013-09-08 18:28:07.000000000 +0200
@@ -74,8 +74,10 @@
 }
 
 -(void)dealloc{
-    [upnpEvents stop];
-	[SSDP stopSSDP];
+    if (upnpEvents)
+        [upnpEvents stop];
+    if (SSDP)
+        [SSDP stopSSDP];
 	[SSDP release];
 	[DB release];
 	[serviceFactory release];
